phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -4006,6 +4037,33 @@ private PhoenixConnection upgradeSystemTask(PhoenixConnection metaConnection)
                         "ALTER TABLE " + taskTableFullName + " SET " + TTL + "=" + TASK_TABLE_TTL);
                 clearCache();
             }
+            // If SYSTEM.TASK does not have disabled regions split policy,
+            // set it up here while upgrading it
+            try (Admin admin = metaConnection.getQueryServices().getAdmin()) {
+                String tableNameStr = PhoenixDatabaseMetaData.SYSTEM_TASK_NAME;
+                TableName tableName;
+                TableDescriptor td;
+                try {
+                    tableName = TableName.valueOf(tableNameStr);
+                    td = admin.getDescriptor(tableName);
+                } catch (org.apache.hadoop.hbase.TableNotFoundException tnfe) {
```

Comment:
Same question here, can we not use namespace mapping property to ensure we are looking for the correct name '.' vs ':'?